### PR TITLE
[dvsim] Add `timescale` to the config field

### PR DESCRIPTION
**Pull request by eunchan**
Revise dvsim `CfgJson.py` to overwrite the default field to the leaf most field if exists. Main usecase is to overwrite `timescale` for a specific test.

Rather than raising `RuntimeError`, it warns now.

### CODE REVIEW

**Review by jdonjdon — APPROVED** [review]

**Comment by eunchan** [comment]
@weicaiyang @cindychip Could you review? I think @jdonjdon needs this to run flash_ctrl test.

File: util/dvsim/CfgJson.py [hunk 1]
```diff
@@ -167,7 +169,13 @@ def set_target_attribute(path, target, key, dict_val):
...
+                'we already had a conflicting value of {!r}.'
+                .format(path, key, dict_val, old_val))
+
     # Neither value looks like a default. Raise an error.
-    raise RuntimeError('{!r}: Value for key {!r} is {!r}, but '
-                       'we already had a conflicting value of {!r}.'
-                       .format(path, key, dict_val, old_val))
+    #raise RuntimeError('{!r}: Value for key {!r} is {!r}, but '
+    #                   'we already had a conflicting value of {!r}.'
+    #                   .format(path, key, dict_val, old_val))
```
Comment (timothytrippel) [inline util/dvsim/CfgJson.py:181]: I think you can just delete these lines, the git history will be there if we need to revert it ever.

**Review by timothytrippel — APPROVED** [review]

File: util/dvsim/CfgJson.py (hunk 1 above)
Comment (eunchan) [reply util/dvsim/CfgJson.py]: Done :)

File: util/dvsim/CfgJson.py [hunk 2]
```diff
@@ -167,7 +169,8 @@ def set_target_attribute(path, target, key, dict_val):
...
 
-    # Neither value looks like a default. Raise an error.
-    raise RuntimeError('{!r}: Value for key {!r} is {!r}, but '
-                       'we already had a conflicting value of {!r}.'
-                       .format(path, key, dict_val, old_val))
+    # both fields are not default. Always uses leaf most field and raise warnings rather than RuntimeError
+    target[key] = old_val
+    log.warning('{!r}: Value for key {!r} is {!r}, but '
+                'we already had a conflicting value of {!r}.'
+                .format(path, key, dict_val, old_val))
```
Comment (eunchan) [inline util/dvsim/CfgJson.py:176]: @sriyerg for future reference.

File: hw/ip/spi_device/dv/spi_device_sim_cfg.hjson [hunk 3]
```diff
@@ -41,6 +41,9 @@
   // Default iterations for all tests - each test entry can override this.
   reseed: 50
 
+  // Overwrite timescale to coarse
+  timescale: "1ns/1ns"
```
Comment (weicaiyang) [inline hw/ip/spi_device/dv/spi_device_sim_cfg.hjson:45]: we use this way to override a field. Here is an [example](https://github.com/lowRISC/opentitan/blob/d9a7c02c1880cc5c8c58f0d51c65003ee3152551/hw/top_earlgrey/dv/chip_sim_cfg.hjson#L52)
```
  overrides: [
    {
      name: timescale
      value: "1ns/1ns"
    }
 ]
```

File: util/dvsim/CfgJson.py [hunk 4]
```diff
@@ -167,7 +169,8 @@ def set_target_attribute(path, target, key, dict_val):
         target[key] = dict_val
         return
 
-    # Neither value looks like a default. Raise an error.
-    raise RuntimeError('{!r}: Value for key {!r} is {!r}, but '
-                       'we already had a conflicting value of {!r}.'
-                       .format(path, key, dict_val, old_val))
+    # both fields are not default. Always uses leaf most field and raise warnings rather than RuntimeError
```
Comment (weicaiyang) [inline util/dvsim/CfgJson.py:172]: we may have multiple level's hjson import. Using override is more clear.

File: hw/ip/spi_device/dv/spi_device_sim_cfg.hjson (hunk 3 above)
Comment (weicaiyang) [reply hw/ip/spi_device/dv/spi_device_sim_cfg.hjson:45]: can you explain where you use 1ns precision?

Comment (eunchan) [reply hw/ip/spi_device/dv/spi_device_sim_cfg.hjson]: Revised to use `overrides`. `1ns` is an example to check if it works or not.

**Review by weicaiyang — APPROVED** [review]

File: util/dvsim/CfgJson.py (hunk 2 above)
Comment (sriyerg) [reply util/dvsim/CfgJson.py]: I don't see this change in the commit. Is it removed now?

**Comment by eunchan** [comment]
Yes. Removed

**Pull request merged**